File: src/016/p016.go
package main

import (
	"fmt"
	"math/big"
)

func bigExponential(input, exponent int64) *big.Int {
	number := big.NewInt(input)
	exp := big.NewInt(exponent)
	mod := big.NewInt(0)
	return number.Exp(number, exp, mod)
}

func solve() int {
	str := bigExponential(2, 1000).String()

	sum := 0
	for i := 0; i < len(str); i++ {
		// str is a string, so str[i]
		// is an ASCII char.
		// -48 is the offset to get
		// zero char as zero integer.
		sum += int(str[i] - 48)
	}

	return sum
}

func main() {
	result := solve()
	fmt.Println(result)
}

File: src/016/p016.lua
function solve()
  local str = string.format("%.0f", 2 ^ 1000)
  local sum = 0
  for i=1, #str do
    local char = str:sub(i, i)
    sum = sum + char
  end
  return math.floor(sum)
end

local result = solve()
print(result)
